core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2596,21 +2597,33 @@ private PhysicalOperation createDynamicFilterSourceOperatorFactory(
                     })
                     .collect(Collectors.toList());
             boolean isReplicatedJoin = isBuildSideReplicated(node);
+            boolean isBuildSideSingle = isBuildSideLocalGatherExchange(node);
```

Comment:
you can use `boolean isBuildSideSingle = context.getDriverInstanceCount().orElse(1) == 1` instead

Reply:
ok, changed to `context.getDriverInstanceCount()`